marlowe-dashboard-client/src/Page/Dashboard/Lenses.purs

```diff
@@ -58,11 +66,14 @@ _tzOffset = _transient <<< prop (Proxy :: _ "tzOffset")
 _templateState :: Lens' State Template.State
 _templateState = _transient <<< prop (Proxy :: _ "templateState")
 
-_runningContracts :: Lens' State (Array ContractState)
-_runningContracts = _derived <<< prop (Proxy :: _ "runningContracts")
+_contracts :: Lens' State (Map MarloweParams ContractState)
+_contracts = _derived <<< prop (Proxy :: _ "contracts")
+
+_closedContracts :: forall r. Monoid r => Fold' r State ContractState
+_closedContracts = _contracts <<< folded <<< filtered _.isClosed
 
-_closedContracts :: Lens' State (Array ContractState)
-_closedContracts = _derived <<< prop (Proxy :: _ "closedContracts")
+_runningContracts :: forall r. Monoid r => Fold' r State ContractState
```

Comment:
I didn't see any performance issues related to this - this should be extremely fast to compute on the fly, filtering an array takes mircoseconds. If we eventually see performance issues and conclude it's because of this, we can decide how we want to optimize at that point, but for now unifying these collections into one and partitioning if / when necessary simplifies the logic quite a bit.

If there is something more likely to cause a performance issue, it is the fact that we recompute the whole execution state every second (once per clock tick) and every wallet update. Probably a better goal would be to reduce the number of rerenders of the Dashboard instead of optimize the render speed. Again though, this hasn't caused any performance problems for us yet, so we can cross that bridge when we get there.